installer/install.iss

```diff
@@ -2024,7 +2058,8 @@ begin
         RefreshProcessList(NIL);
         Result:=(GetArrayLength(Processes)=0);
     end else begin
-        Result:=False;
+        // skip pages that are entirely preset
+        Result:=(Pos(','+IntToStr(PageID)+',',PagesPreSet)>0);
```

Comment:
Whoa. This change would _always_ skip the options pages when upgrading, not only in some sort of "easy mode"...

I think we _have_ to guard this behind an initial page that lets the user choose between the easy and the advanced installation.

Further, I am not entirely sure that all of our pages are handled correctly: at least the component page, and probably also the page that lets the user choose whether to run Git Bash and whether to show the release notes, would probably be skipped by this patch? (I don't know, I did not build an installer with this patch, did you?)

Reply:
> Further, I am not entirely sure that all of our pages are handled correctly: at least the component page, and probably also the page that lets the user choose whether to run Git Bash and whether to show the release notes, would probably be skipped by this patch? (I don't know, I did not build an installer with this patch, did you?)

I left the built-in pages with no changes, which causes the license and components pages to be shown on upgrade.  We could figure out how to skip these pages as well if you would prefer that.  [Aside: One thing I notice is that many of the items on the components page seem more like options than components, and it might make sense to have them as custom options page(s) instead?]

Reply:
> I left the built-in pages with no changes, which causes the license and components pages to be shown on upgrade. We could figure out how to skip these pages as well if you would prefer that.

I think that would actually make a lot of sense. The license has not changed, the user accepted it already (by virtue of installing the software), and the same thing for options.

Except for that hard problem of new options...

> [Aside: One thing I notice is that many of the items on the components page seem more like options than components, and it might make sense to have them as custom options page(s) instead?]

I think you're right. The Git LFS option looks somewhat like it is a component, although I wouldn't mind turning that into a proper option.